### PR TITLE
Design Documentation and Code Improvements

### PR DESCRIPTION
**Pull request by mpatrou**
It includes the design documentation and some additional improvements.

# Short description of the changes:

- documentation: class, MVP, workflows, etc
- PySide6 issue fixed
- src/class renaming

# Long description of the changes:

# Check list for the pull request
- [ ] I have read the [CONTRIBUTING]
- [ ] I have read the [CODE_OF_CONDUCT]
- [ ] I have added tests for my changes
- [ ] I have updated the documentation accordingly

# Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

# Manual test for the reviewer

# References

### CODE REVIEW

File: docs/source/design/hyspecpptclasses.rst [hunk 1]
```diff
@@ -1,56 +1,268 @@
...
-        +calculate_qmod()
-        +get_single_crystal_data()
+    class SingleCrystalParameters{
+        +float lattice_a
+        +float lattice_b
+        +float lattice_c
+        +float lattice_alpha
+        +float lattice_beta
+        +float lattice_gamma
+        +str lattice unit_h
```
Comment (AndreiSavici) [inline docs/source/design/hyspecpptclasses.rst]: h, k, l are floats

File: docs/source/design/hyspecpptclasses.rst [hunk 2]
```diff
@@ -1,56 +1,268 @@
...
-        +Double detector_tank_angle_s
-        +Double polarization_direction_angle_p
-        +Double delta_e
-        +Double mod_q
-        +Option graph_type
+        +Sample sample
+    }
+
+    class Sample{
+        +str current_sample_type
```
Comment (AndreiSavici) [inline docs/source/design/hyspecpptclasses.rst]: Should sample type and graph type be some enums?

File: docs/source/design/hyspecpptclasses.rst [hunk 3]
```diff
@@ -1,56 +1,268 @@
...
+
+     .. code-block:: bash
+
+        {
+            "q_min": [], //1-d array
+            "q_max": [], //1-d array
+            "energy_transfer" : [], //1-d array
+            "q2d" :[[],], //2-d array
+            "e2d" :[[],], //2-d array
+            "scharf_angle" :[[],], //2-d array
```
Comment (AndreiSavici) [inline docs/source/design/hyspecpptclasses.rst]: scharpf

File: docs/source/design/hyspecpptclasses.rst [hunk 4]
```diff
@@ -1,56 +1,268 @@
...
+            "delta_e": <d_e>,
+            "mod_q" : <m_q>,
+            "graph_type" : <g_a>,
+            "sc_parametes" : {}
+        }
+
+
+* get_emin(delta_e, incident_energy_e) --> float : The function returns the e_min value, based on delta_e and incident_energy_e. If delta_e < -incident_energy_e, then e_min =1.2* delta_e, else e_min = delta_e.
+* calculate_qmod() --> float :  The function returns qmod. It calculates the value from the sc_parametes (SingleCrystal mode).
+* calculate_crosshair() --> dict : The function calculates the crosshair x and y float values from the sc_parametes (SingleCrystal mode). The following format is returned:
```
Comment (AndreiSavici) [inline docs/source/design/hyspecpptclasses.rst]: These are mod_q and delta_e both for powder and single crystal

**Review by AndreiSavici — CHANGES_REQUESTED** [review]
Minor changes

File: docs/source/design/hyspecpptclasses.rst [hunk 5]
```diff
@@ -1,95 +1,317 @@
...
+                "lattice_unit_l":<l>
+            }
+        }
+
+    Internally store_data() is called to store the parameters, and get_qmod() is called to find qmod values respectively.
+    The following format is returned:
+
+     .. code-block:: bash
+
+        {
```
Comment (AndreiSavici) [inline docs/source/design/hyspecpptclasses.rst:158]: Both eline and qline are single float values. Transformation to 2 element list happens in view

File: docs/source/design/hyspecpptclasses.rst [hunk 6]
```diff
@@ -108,13 +330,97 @@ HyspecPPT View
...
 
+Functions
+-------------
+
+The function signatures and description are included below.
+
+**-- HyspecPPTView**
+
+* def update_plot(q_min: list[float],q_max: list[float],energy_transfer: list[float], q2d: list[list[float]],e2d: list[list[float]], scharpf_angle: list[list[float]]) --> None : The function updates the plot with the given parameters.
+* def update_crosshair(eline: list[float], qline:list[float]) --> None : The function updates the crosshair lines at the plot with the given parameters.
```
Comment (AndreiSavici) [inline docs/source/design/hyspecpptclasses.rst]: See comment above. Should use floats. Transformation to list is internal here

File: docs/source/design/hyspecpptclasses.rst [hunk 7]
```diff
@@ -136,3 +446,205 @@ HyspecPPT Presenter
...
+    * Invalid Status:
+
+    .. mermaid::
+
+        sequenceDiagram
+            participant View
+            participant Presenter
+            participant Model
+
+            Note over View,Model: Crosshair update due to any ExperimentWidget parameter update
```
Comment (AndreiSavici) [inline docs/source/design/hyspecpptclasses.rst]: Note is wrong (it should be the same as the valid status

**Review by AndreiSavici — CHANGES_REQUESTED** [review]
Minor changes required

File: docs/source/design/hyspecpptclasses.rst (hunk 7 above)
Comment (KyleQianliMa) [reply docs/source/design/hyspecpptclasses.rst]: Which line do you mean should be the same as valid status? Line 535? or the entire note?

Comment (AndreiSavici) [reply docs/source/design/hyspecpptclasses.rst]: "Crosshair update due to any ExperimentWidget parameter update" should be "Plot draw due to any ExperimentWidget parameter update"

**Review by AndreiSavici — APPROVED** [review]

**Pull request merged**